### PR TITLE
Private IP Disclosure Handle Non-Standard Ports

### PR DESCRIPTION
**Pull request by kingthorin**
TestInfoPrivateAddressDisclosure - Change the comparison of private IP matches to only compare the IP or Host portion (excluding port).

TestInfoPrivateAddressDisclosureUnitTest - Add an additional test which covers the revised behavior both with and without port being specified.

Fixes zaproxy/zaproxy#3549

### CODE REVIEW

File: src/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosure.java [hunk 1]
```diff
@@ -108,7 +108,7 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         
         while (matcher.find()) {
             String privateIp = matcher.group();
-            if (getLevel() != AlertThreshold.LOW && privateIp.equalsIgnoreCase(host)) {
+            if (getLevel() != AlertThreshold.LOW && privateIp.toLowerCase().contains(host)) {
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosure.java]: It could be used `matcher.group(1)` for this check, which returns just the private IP. Otherwise the `host` should be lowered case (and used `Locale.ROOT`), worth adding a test for mixed case?

File: test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java [hunk 2]
```diff
@@ -227,6 +228,32 @@ public void alertsIfPrivIpAndDropsPortNoInEvidence() throws HttpMalformedHeaderE
 			}
 		}
 	}
+	
+	@Test
+	public void shouldNotAlertIfPrivIpIsTarget() throws HttpMalformedHeaderException, URIException {
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java]: Could this test be moved closer to related tests? Or update the existing test (`shouldIgnoreRequestedPrivateIpByDefault`) that checks the same?

File: test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java [hunk 3]
```diff
@@ -227,6 +228,32 @@ public void alertsIfPrivIpAndDropsPortNoInEvidence() throws HttpMalformedHeaderE
...
+				":45876",
+				":8081",
+				":98", 
+				""
+		};
+		for (int pi = 0; pi < ports.length; pi++) {
+			for (int ii = 0; ii < ipHost.length; ii++) {
+				alertsRaised.clear();
+				String candidate = ipHost[ii] + ports[pi];
+				HttpMessage msg = createHttpMessage(candidate);
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java]: It could use `createHttpMessage(String, String)`, instead of setting the URI again.

**Comment by kingthorin** [comment]
All fixed

File: src/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosure.java [hunk 4]
```diff
@@ -108,7 +109,7 @@ public void scanHttpResponseReceive(HttpMessage msg, int id, Source source) {
         
         while (matcher.find()) {
             String privateIp = matcher.group();
```
Comment (thc202) [inline src/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosure.java]: This variable could be dropped (just used in one place now).

File: test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java [hunk 5]
```diff
@@ -90,15 +90,27 @@ public void alertsIfPrivateIp() throws HttpMalformedHeaderException {
 	}
 
 	@Test
-	public void shouldIgnoreRequestedPrivateIpByDefault() throws HttpMalformedHeaderException {
+	public void shouldIgnoreRequestedPrivateIpByDefault() throws HttpMalformedHeaderException, URIException, NullPointerException {
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java]: Could declare that throws Exception instead.

File: test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java [hunk 6]
```diff
@@ -90,15 +90,27 @@ public void alertsIfPrivateIp() throws HttpMalformedHeaderException {
...
 		// Given
-		String privateIp = "192.168.36.127";
-		String requestUri = "https://" + privateIp + ":8123/";
-		HttpMessage msg = createHttpMessage(requestUri, privateIp);
-		// When
-		rule.scanHttpResponseReceive(msg, -1, createSource(msg));
-		// Then
-		assertThat(alertsRaised, is(empty()));
+		// ip and aws-hostname which get concatenated with the ports as candidates
+		String[] ipHost = new String[] {"10.0.2.2", "iP-10-0-2-2"};
```
Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java]: The regular expression is not case insensitive, not sure worth changing it(?) In any case we need to change the case of the request URI (to have them with different case).

Comment (kingthorin) [inline test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java]: I'm going to revert this change actually, per `passesIfInvalidAwsHostname()` is seems like it should be lower only.

Comment (thc202) [inline test/org/zaproxy/zap/extension/pscanrules/TestInfoPrivateAddressDisclosureUnitTest.java]: OK.

**Comment by thc202** [comment]
Thanks!
